lib/simplifyExpression/collectAndCombineSearch/LikeTermCollector.js

```diff
@@ -123,11 +123,14 @@ LikeTermCollector.collectLikeTerms = function(node) {
 // returns this group 'name'
 function getTermName(node, termSubclass, op) {
   const term = new termSubclass(node);
+  const baseNode = term.getBaseNode()
   // we 'name' terms by their base node name
-  let termName = print.ascii(term.getBaseNode());
+  // #dent: assume symbol node to avoid costly printAscii function
+  let termName = baseNode.name || print.ascii(baseNode);
   // when adding terms, the exponent matters too (e.g. 2x^2 + 5x^3 can't be combined)
   if (op === '+') {
-    const exponent = print.ascii(term.getExponentNode(true));
+    exponentNode = term.getExponentNode(true)
+    const exponent = term.value || term.name || print.ascii(exponentNode);
```

Comment:
basically, if it's a single node like a variable node (that just has a name) or a constant node (that just has a value), we can save this ourselves this nasty print operation